python/pyspark/ml/param/__init__.py

```diff
@@ -389,8 +391,8 @@ def copy(self, extra=None):
         if extra is None:
             extra = dict()
         that = copy.copy(self)
-        that._paramMap = self.extractParamMap(extra)
-        return that
+        that._paramMap = {}
```

Comment:
The reason I changed this is because when you pass in "extra" params, they will get set in the param map without ever calling `_set`. This is particularly a problem when people use param maps for cross validator, which passes "extra" params through the `fit` call. So, if someone specifies an int for a param that takes float, they get the nasty py4j error.